test-e2e/first-hit.test.ts

```diff
@@ -0,0 +1,121 @@
+import { test, expect } from './fixtures/config-test-fixtures.js'
+import { waitForServiceWorker } from './fixtures/wait-for-service-worker.js'
+
+test.describe('first-hit ipfs-hosted', () => {
+  /**
+   * "ipfs-hosted" tests verify that when the _redirects is hit and redirects to the <root>?helia-sw=<path> that navigation is handled correctly.
```

Comment:
since we don't have an ipfs server to test this with yet, and we don't need that complexity, we just test that the redirect path is handled properly.

eventually, we would test the full flow to guarantee ipfs hosting works e2e, which could be a good example to show users how to host this on ipfs themselves.